File: dataStructures/queue/javascript/queue.js
class Queue {
  constructor() {
    this.items = [];
    this.is_empty.bind = this.is_empty;
    this.push.bind = this.push;
    this.pop.bind = this.pop;
    this.print_queue.bind = this.print_queue;
  }

  // check if it's empty
  is_empty() {
    return this.items.length === 0;
  }

  // add element
  push(item) {
    this.items.unshift(item);
  }

  // remove element
  pop() {
    if (this.is_empty()) {
      console.log("There is nothing to delete!");
      return;
    }
    this.items.pop();
  }

  // output element
  print_queue() {
    console.log(this.items);
  }
}

module.exports = Queue;

File: dataStructures/queue/python/queue.py
class Queue:
    def __init__(self):
        self.items = []

    # check if the stack is empty or not
    def is_empty(self):
        return len(self.items) == 0

    # push a new element onto the stack
    def push(self, item):
        self.items.insert(0, item)

    # pop an element from the stack
    def pop(self):
        if self.is_empty():
            print("There is no element to remove")
            return
        return self.items.pop()

    # print the stack
    def print_queue(self):
        print(self.items)
